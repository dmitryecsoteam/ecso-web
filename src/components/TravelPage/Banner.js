import React from 'react';
import classNames from 'classnames';
import TrackVisibility from 'react-on-screen';

export default ({ linkTo, backgroundImage, textMain, textSecondary, textButton }) => {

    const BannerVisible = ({ isVisible }) => {
        const className = classNames('banner__container', { 'banner__container--animation': isVisible });
        return (
            <div className={className}>
                <a
                    className="banner__link"
                    href={linkTo}
                >
                    <div
                        className="banner__image"
                        style={{ backgroundImage: `url(${backgroundImage})` }}
                    >
                    </div>
                    <span className="banner__text banner__text--main">{textMain}</span>
                    <span className="banner__text banner__text--secondary">{textSecondary}</span>
                    <div className="banner__button">{textButton}</div>
                </a>
            </div>
        );
    };

    return (
        <TrackVisibility once className="banner">
            <BannerVisible />
        </TrackVisibility>
    );
}