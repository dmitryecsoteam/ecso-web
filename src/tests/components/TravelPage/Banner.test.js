import React from 'react';
import { mount } from 'enzyme';
import Banner from '../../../components/TravelPage/Banner';

test('should render Banner component', () => {
    const wrapper = mount(<Banner
        linkTo="#"
        backgroundImage="test image"
        textMain="main text here"
        textSecondary="secondary text here"
        textButton="button"
    />);
    expect(wrapper).toMatchSnapshot();
});