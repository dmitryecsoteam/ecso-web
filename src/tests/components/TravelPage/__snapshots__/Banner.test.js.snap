// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Banner component 1`] = `
<_default
  backgroundImage="test image"
  linkTo="#"
  textButton="button"
  textMain="main text here"
  textSecondary="secondary text here"
>
  <TrackVisibility
    className="banner"
    offset={0}
    once={true}
    partialVisibility={false}
    tag="div"
    throttleInterval={150}
  >
    <div
      className="banner"
    >
      <BannerVisible
        isVisible={false}
        key=".0"
      >
        <div
          className="banner__container"
        >
          <a
            className="banner__link"
            href="#"
          >
            <div
              className="banner__image"
              style={
                Object {
                  "backgroundImage": "url(test image)",
                }
              }
            />
            <span
              className="banner__text banner__text--main"
            >
              main text here
            </span>
            <span
              className="banner__text banner__text--secondary"
            >
              secondary text here
            </span>
            <div
              className="banner__button"
            >
              button
            </div>
          </a>
        </div>
      </BannerVisible>
    </div>
  </TrackVisibility>
</_default>
`;
